src/mock.utils.ts

```diff
@@ -87,21 +92,68 @@ export class MockEthersProvider {
     return this;
   }
 
-  public addFilteredLogs(filter: Filter | FilterByBlockHash, logs: Log[]): MockEthersProvider {
-    const matcher = {
-      ...filter,
-      topics: (filter.topics)
-        ? expect.arrayContaining(filter.topics.map(el => (Array.isArray(el))? expect.arrayContaining(el) : el))
-        : undefined,
-    };
+  public addLogs(...logs: Log[]): MockEthersProvider {
+    this.logs.push(...logs);
+    return this;
+  }
+
+  private _getLogs(filter: Filter | FilterByBlockHash): Log[] {
+    let logs = this.logs;
+
+    if (filter.address) logs = logs.filter((log) => log.address === filter.address);
```

Comment:
should we add `.toLowerCase()` when comparing in both members just to avoid fake mismatches?

Reply:
sure, good idea